### PR TITLE
#125: Pass column types to CSV iterator

### PR DESCRIPTION
**Pull request by kaklakariada**
This is part of #125
Also delete unused classes.

### CODE REVIEW

File: src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvIterator.java [hunk 1]
```diff
@@ -2,55 +2,58 @@
...
 
 /**
- * This class iterates the lines of a JSON-Lines file an creates for each line a JSON {@link DocumentNode}.
+ * This class iterates the lines of a CSV file and creates a {@link CsvObjectNode} for each line.
  */
 class CsvIterator implements CloseableIterator<DocumentNode> {
     // efficient reading of characters, arrays and lines
     private final InputStreamReader inputStreamReader;
-    private long lineCounter = 0;
+    private final List<ColumnMapping> csvColumns;
```
Comment (ckunki) [inline src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvIterator.java:24]: Is the field `csvColumns` unused by intention?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvIterator.java (hunk 1 above)
Comment (kaklakariada) [reply src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvIterator.java:24]: Yes. It will be used soon.

**Review by ckunki — APPROVED** [review]

File: src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/ConvertingCsvIterator.java [hunk 2]
```diff
@@ -2,19 +2,18 @@
...
+class ConvertingCsvIterator<T> implements Iterator<DocumentNode> {
 
-class NamedCsvIterator implements Iterator<DocumentNode> {
+    private final Iterator<T> delegate;
+    private final Function<T, DocumentNode> converter;
 
-    private final Iterator<NamedCsvRow> delegate;
-
-    NamedCsvIterator(final NamedCsvReader namedCsvReader) {
+    ConvertingCsvIterator(final Iterable<T> namedCsvReader, final Function<T, DocumentNode> converter) {
```
Comment (ckunki) [inline src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/ConvertingCsvIterator.java:14]: rename  `namedCsvReader` to `delegate`?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvIterator.java [hunk 3]
```diff
@@ -46,17 +43,17 @@ static CsvIterator create(final RemoteFile csvFile, final List<ColumnMapping> cs
...
     private static Iterator<DocumentNode> createDelegate(final CsvConfiguration csvConfiguration,
-            final InputStreamReader inputStreamReader) {
+            final CsvObjectNodeFactory nodeFactory, final InputStreamReader inputStreamReader) {
         if (hasHeaders(csvConfiguration)) {
-            return new ConvertingCsvIterator<>(NamedCsvReader.builder().build(inputStreamReader),
-                    NamedCsvObjectNode::new);
+            return new ConvertingCsvIterator<>(NamedCsvReader.builder().build(inputStreamReader), nodeFactory::create);
         } else {
-            return new ConvertingCsvIterator<>(CsvReader.builder().build(inputStreamReader), CsvObjectNode::new);
+            return new ConvertingCsvIterator<>(CsvReader.builder().build(inputStreamReader), nodeFactory::create);
```
Comment (ckunki) [inline src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvIterator.java:56]: 👍

**Review by ckunki — APPROVED** [review]

**Pull request merged**